### PR TITLE
Add parameters order format

### PR DESCRIPTION
**Pull request by MassimoPolimeni**
This PR adds **only** 1.14 point about **formatting the parameters order**; I know this is not a simple topic because we could have totally different view on this.

What you will read is just my proposal, based on my honest opinion and some thoughts from part of the iOS team. Happy to discuss it, any suggestions or comments are appreciated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
 	flowLayout.sectionInset.left
 ```
 
+* **1.14** Properties should be ordered following logical order; in order to have a good readibility we want to group them in three main sets representing the visibility scope:
+1. public
+2. internal
+3. private
+
+and keeping the **computer properties** at the bottom of them.
```
Comment (mugx-fc) [inline README.md]: computed

File: README.md [hunk 2]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+public var variable3: ...
+public var variable4: ...
+public var variable5: ...
+public var variable6: ...
+public var variable7: ...
+public var variable8: ...
+public var variable9: ...
+public var variable10: ...
+
+internal var ...: ... // less than 10 elements in this group, keep them logically ordered
```
Comment (AdamM-FC) [inline README.md]: Maybe should clarify, if elements should be ordered logically if possible, e.g. model storing user details

```
let firstName: String
let lastName: String
let email: String
```

However, if there is no clear order in the model, then the default organising should be alphabetical order

File: README.md [hunk 3]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+var name: String
+var surname: String
+var ...: ...
+var fullName: String { // keep computed properties at the end of the group
```
Comment (AdamM-FC) [inline README.md]: Looking at the rest of the style guide, comments should be above the line of code so for consistency move this above the computed property

File: README.md [hunk 4]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+    return name + surname
+}
+
+private var age: Int?
+private var spinner: UIActivityIndicatorView? { return ... } // keep computed properties at the end of the group
+
+// NOT PREFERRED
+public var variable1: ...
```
Comment (AdamM-FC) [inline README.md]: These variables are still in alphabetical order, so looks a bit confusing having this under not preferred

File: README.md [hunk 5]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+public var variable2: ...
+public var variable3: ...
+public var variable5: ...
+public var variable4: ...
+public var variable7: ...
+public var variable8: ...
+public var variable9: ...
+public var variable6: ...
+public var variable10: ... // not  preferring alphabetical order
```
Comment (AdamM-FC) [inline README.md]: You already have NOT PREFERRED above, don't need to specify this again

File: README.md [hunk 6]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+
+internal var ...: ...
+var name: String
+var surname: String
+private var ...: ... // private properties should be grouped
```
Comment (AdamM-FC) [inline README.md]: ditto about comment being above the line of code

File: README.md [hunk 7]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+var fullName: String { // computed properties should be at the end of the group
```
Comment (AdamM-FC) [inline README.md]: ditto about comment being above the line of code

File: README.md [hunk 8]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+    return name + surname
+}
+var ...: ...
+
+private var spinner: UIActivityIndicatorView? { return ... } // computed properties should be at the end of the group
```
Comment (AdamM-FC) [inline README.md]: ditto about comment being above the line of code

File: README.md [hunk 9]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+public var variable3: ...
+public var variable5: ...
+public var variable4: ...
+public var variable7: ...
+public var variable8: ...
+public var variable9: ...
+public var variable6: ...
+public var variable10: ... // not  preferring alphabetical order
+
+internal var ...: ...
```
Comment (AdamM-FC) [inline README.md]: should remove lines of code like this, isn't clear what it's purpose is

File: README.md [hunk 10]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
 	flowLayout.sectionInset.left
 ```
 
+* **1.14** Properties should be ordered following logical order; in order to have a good readibility we want to group them in three main sets representing the visibility scope:
+1. public
+2. internal
+3. private
+
+and keeping the **computed properties** at the bottom of them.
+An **empty line** should always separate these groups; in addition if one of this group of properties counts **10 or more elements** or there is no clear logic order than we should keep them ordered **alphabetically** just in that group.
```
Comment (0mega) [inline README.md]: This renders strangely, the text after the three numbered bullets is a continuation of the unordered bullet at the top, but it is formatted as completely separate paragraph.
Maybe worth starting new sentence or creating new bullet 🤷‍♂

File: README.md [hunk 11]
```diff
@@ -179,6 +179,63 @@ let cellWidth = collectionView.bounds.size.width +
...
+public var variable1: ...
+public var variable2: ...
+public var variable3: ...
+public var variable4: ...
+public var variable5: ...
+public var variable6: ...
+public var variable7: ...
+public var variable8: ...
+public var variable9: ...
+public var variable10: ...
```
[the remark this answers is not in the document]
Comment (MassimoPolimeni) [reply README.md]: just to underline we want to keep the alphabetic order when we have 10 or more elements (in addition if we don't have any logical order) 😃

**Pull request closed without merging**